python-backend/app/api/permits/permit/resources/permit.py

```diff
@@ -118,24 +98,73 @@ def post(self, permit_guid=None):
                 document_manager_guid=newFile['document_manager_guid'],
                 mine_guid=permit.mine_guid,
             )
-            amendment.documents.append(new_pa_doc)
+            amendment.related_documents.append(new_pa_doc)
         db.session.commit()
 
         permittee = MinePartyAppointment.create(
             mine.mine_guid, data.get('permittee_party_guid'), 'PMT', datetime.utcnow(), None,
             self.get_user_info(), permit.permit_guid, True)
         db.session.commit()
 
-        return permit.json()
+        return permit
+
+
+class PermitResource(Resource, UserMixin):
+    parser = reqparse.RequestParser(trim=True)
+    parser.add_argument(
+        'permit_no', type=str, help='Number of the permit being added.', location='json')
+    parser.add_argument('mine_guid', type=str, help='GUID of the mine.', location='json')
+    parser.add_argument(
+        'permittee_party_guid',
+        type=str,
+        help='GUID of the party that is the permittee for this permit.',
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'permit_status_code',
+        type=str,
+        location='json',
+        help='Status of the permit being added.',
+        store_missing=False)
+    parser.add_argument(
+        'received_date',
+        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'issue_date',
+        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'authorization_end_date',
+        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'permit_amendment_status_code',
+        type=str,
+        location='json',
+        help='Status of the permit being added.',
+        store_missing=False)
+    parser.add_argument(
+        'description', type=str, location='json', help='Permit description', store_missing=False)
+    parser.add_argument('uploadedFiles', type=list, location='json', store_missing=False)
 
     @api.doc(params={'permit_guid': 'Permit guid.'})
-    @requires_role_mine_create
-    def put(self, permit_guid=None):
-        if not permit_guid:
-            raise BadRequest('Permit guid was not provided.')
+    @requires_role_mine_view
+    @api.marshal_with(PERMIT_MODEL, code=200)
+    def get(self, permit_guid):
+        permit = Permit.find_by_permit_guid_or_no(permit_guid)
+        if not permit:
+            raise NotFound('Permit not found.')
+        return permit
 
+    @api.doc(params={'permit_guid': 'Permit guid.'})
+    @requires_role_mine_create
+    @api.marshal_with(PERMIT_MODEL, code=200)
+    def put(self, permit_guid):
```

Comment:
when this path changes, mine_guid will need to be checked against the permit

Reply:
Path changed, and mine_guid's are validated. After working with it I don't like this method of longer paths and strict validation. It adds cruft. The front-end components need to pass more things through, it adds extra SQL calls and overall complexity that I don't think will be offset by this notion of extra safety.